plugins/modules/system/iptables_state.py

```diff
@@ -0,0 +1,638 @@
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: iptables_state
+short_description: Save iptables state into a file or restore it from a file
+version_added: "2.10"
+author: quidame (@quidame)
+description:
+  - C(iptables) is used to set up, maintain, and inspect the tables of IP
+    packet filter rules in the Linux kernel.
+  - This module handles the saving and/or loading of rules. This is the same
+    as the behaviour of the C(iptables-save) and C(iptables-restore) (or
+    C(ip6tables-save) and C(ip6tables-restore) for IPv6) commands which this
+    module uses internally.
+  - Modifying the state of the firewall remotely may lead to loose access to
+    the host in case of mistake in new ruleset. This module embeds a rollback
+    feature to avoid this, by telling the host to restore previous rules if a
+    cookie is still there after a given delay, and all this time telling the
+    controller to try to remove this cookie on the host through a new
+    connection.
+notes:
+  - The rollback feature is not a module option and depends on task's
+    attributes. To enable it, the module must be played asynchronously, i.e.
+    by setting task attributes I(poll) to I(0), and I(async) to a value less
+    or equal to C(ANSIBLE_TIMEOUT). If I(async) is greater, the rollback will
+    still happen if it shall happen, but you will experience a connection
+    timeout instead of more relevant info returned by the module after its
+    failure.
+options:
+  counters:
+    description:
+      - Save or restore the values of all packet and byte counters.
+      - When I(True), the module is not idempotent.
+    type: bool
+    default: false
+  ip_version:
+    description:
+      - Which version of the IP protocol this module should apply to.
+    type: str
+    choices: [ ipv4, ipv6 ]
+    default: ipv4
+  modprobe:
+    description:
+      - Specify the path to the modprobe program internally used by iptables
+        related commands to load kernel modules.
+      - By default, /proc/sys/kernel/modprobe is inspected to determine the
+        executable's path.
+    type: path
+  noflush:
+    description:
+      - For I(state=restored), ignored otherwise.
+      - If I(False), restoring iptables rules from a file flushes (deletes)
+        all previous contents of the respective table(s). If I(True), the
+        previous rules are left untouched (but policies are updated anyway,
+        for all built-in chains).
+    type: bool
+    default: false
+  path:
+    description:
+      - The file the iptables state should be saved to.
+      - The file the iptables state should be restored from.
+      - Required when I(state=saved) or I(state=restored).
```

Comment:
You mean, it is **always** required as there are no other states?

Reply:
no, I meant 'state' and 'path' are required together, that is no more the case since stateless/informative behaviour has been dropped just now.